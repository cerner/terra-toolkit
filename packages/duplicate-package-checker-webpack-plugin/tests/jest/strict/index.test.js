const webpack = require('webpack');
const assert = require('assert');
const chalk = require('chalk');
const MakeConfig = require('./make.webpack.config');

const chalkLevel = chalk.level;

describe('Simple dependency tree', () => {
  beforeEach(() => {
    /**
     * Travis-ci doesn't support colorized string generation with Chalk. To prevent snapshot mismatches
     * with Travis's test output, Chalk is disabled for snapshot tests.
     */
    chalk.level = 0;
  });

  afterEach(() => {
    chalk.level = chalkLevel;
  });

  it('should output warnings if strict', (done) => {
    webpack(MakeConfig(), (err, stats) => {
      assert(stats.compilation.warnings.length === 2);
      expect(stats.compilation.warnings[0].message).toMatchSnapshot();
      expect(stats.compilation.warnings[1].message).toMatchSnapshot();
      done();
    });
  });

  it('should not output warnings if not strict', (done) => {
    webpack(
      MakeConfig({
        strict: false,
      }),
      (err, stats) => {
        assert(stats.compilation.warnings.length === 1);
        expect(stats.compilation.warnings[0].message).toMatchSnapshot();
        done();
      },
    );
  });
});
