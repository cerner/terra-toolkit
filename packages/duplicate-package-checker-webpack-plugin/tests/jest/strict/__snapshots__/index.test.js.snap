// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple dependency tree should not output warnings if not strict 1`] = `
"a
  Multiple versions of a found:
    2.0.0 ./~/b/~/a
    2.5.0 ./~/c/~/a

Check how you can resolve duplicate packages: 
https://github.com/cerner/terra-toolkit/tree/main/packages/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle
"
`;

exports[`Simple dependency tree should output warnings if strict 1`] = `
"a
  Multiple versions of a found:
    1.0.0 ./~/a
    2.0.0 ./~/b/~/a
    2.5.0 ./~/c/~/a
"
`;

exports[`Simple dependency tree should output warnings if strict 2`] = `
"b
  Multiple versions of b found:
    1.0.0 ./~/b
    2.0.0 ./~/c/~/a/~/b

Check how you can resolve duplicate packages: 
https://github.com/cerner/terra-toolkit/tree/main/packages/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle
"
`;
